fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -765,6 +739,201 @@ private ParameterDAO makeParameterDAO(Connection connection) throws FHIRPersiste
+        // search resource type to returned logical resource IDs. The logical resource IDs
+        // are used in the include queries.
+        Map<Integer, Map<String, Set<String>>> queryResultMap = new HashMap<>();
+        
+        // Add base query result to map
+        String resourceTypeString = resourceType.getSimpleName();
+        Set<String> baseLogicalResourceIds = resourceDTOList.stream()
+                .map(r -> Long.toString(r.getLogicalResourceId())).collect(Collectors.toSet());
+        queryResultMap.put(0, Collections.singletonMap(resourceTypeString, baseLogicalResourceIds));
+
+        // Process non-iterative _include parameters. These are only run against 'match' search
+        // results.
+        for (InclusionParameter includeParm : searchContext.getIncludeParameters()) {
+            if (!includeParm.isIterate()) {
+                // Build and run the query
+                List<com.ibm.fhir.persistence.jdbc.dto.Resource> includeResources =
+                        this.runIncludeQuery(resourceType, searchContext, queryBuilder, includeParm, SearchConstants.INCLUDE,
+                            baseLogicalResourceIds, queryResultMap, resourceDao, 1, allResourceIds);
+                
+                // Add new ids to de-dup list
+                allResourceIds.addAll(includeResources.stream().map(r -> r.getId()).collect(Collectors.toSet()));
+
+                // Add resources to list
+                allIncludeResources.addAll(includeResources);
+                
+                // Check if max size exceeded. If so, return results and let rest helper throw exception.
+                if (allIncludeResources.size() > SearchConstants.MAX_PAGE_SIZE) {
+                    return allIncludeResources;
+                }
+            }
+        }
+
+        // Process non-iterative _revinclude parameters. These are only run against 'match' search
+        // results.
+        for (InclusionParameter revincludeParm : searchContext.getRevIncludeParameters()) {
+            if (!revincludeParm.isIterate()) {
+                // Build and run the query
+                List<com.ibm.fhir.persistence.jdbc.dto.Resource> revincludeResources =
+                        this.runIncludeQuery(resourceType, searchContext, queryBuilder, revincludeParm, SearchConstants.REVINCLUDE,
+                            baseLogicalResourceIds, queryResultMap, resourceDao, 1, allResourceIds);
+
+                // Add new ids to de-dup list
+                allResourceIds.addAll(revincludeResources.stream().map(r -> r.getId()).collect(Collectors.toSet()));
+
+                // Add resources to list
+                allIncludeResources.addAll(revincludeResources);
+                
+                // Check if max size exceeded. If so, return results and let rest helper throw exception.
+                if (allIncludeResources.size() > SearchConstants.MAX_PAGE_SIZE) {
```

Comment:
Just a minor concern about signaling the error condition here. If the rest helper uses different logic to determine the error case, it could be tricky to track down. I don't know if there's a cleaner way.

Reply:
Agree with your concern, but I think this is the most straight-forward way to handle it.